includes/class-pirateforms.php

```diff
@@ -0,0 +1,308 @@
+
+		$this->loader->add_action( 'wp_enqueue_scripts', $plugin_public, 'enqueue_styles_and_scripts' );
+		$this->loader->add_action( 'template_redirect', $plugin_public, 'pirate_forms_process_contact' );
+		$this->loader->add_action( 'wp_ajax_pirate_forms_save', $plugin_public, 'pirate_forms_save_callback' );
+		$this->loader->add_action( 'wp_ajax_nopriv_pirate_forms_save', $plugin_public, 'pirate_forms_save_callback' );
+		$this->loader->add_filter( 'widget_text', $plugin_public, 'pirate_forms_widget_text_filter', 9 );
+		$this->loader->add_action( 'init', $this, 'pirate_forms_register_content_type' );
+
+		add_shortcode( 'pirate_forms', array( $plugin_public, 'pirate_forms_display_form' ) );
+	}
+
+	/**
+	 * Run the loader to execute all of the hooks with WordPress.
+	 *
+	 * @since    1.0.0
+	 */
+	public function run() {
+		add_action( 'widgets_init', create_function( '', 'return register_widget("pirate_forms_contact_widget");' ) );
+
+		$this->loader->run();
+	}
+
+	/**
+	 * The name of the plugin used to uniquely identify it within the context of
+	 * WordPress and to define internationalization functionality.
+	 *
+	 * @since     1.0.0
+	 * @return    string    The name of the plugin.
+	 */
+	public function get_plugin_name() {
+		return $this->plugin_name;
+	}
+
+	/**
+	 * The reference to the class that orchestrates the hooks with the plugin.
+	 *
+	 * @since     1.0.0
+	 * @return    PirateForms_Loader    Orchestrates the hooks of the plugin.
+	 */
+	public function get_loader() {
+		return $this->loader;
+	}
+
+	/**
+	 * Retrieve the version number of the plugin.
+	 *
+	 * @since     1.0.0
+	 * @return    string    The version number of the plugin.
+	 */
+	public function get_version() {
+		return $this->version;
+	}
+
+	/**
+	 * Returns if the domain is localhost
+	 *
+	 * @since     1.0.0
+	 */
+	public static function pirate_forms_is_localhost() {
+		$server_name = strtolower( $_SERVER['SERVER_NAME'] );
+		return in_array( $server_name, array( 'localhost', '127.0.0.1' ) );
+	}
+
+	/**
+	 * Gets the from email
+	 *
+	 * @since     1.0.0
+	 */
+	public static function pirate_forms_from_email() {
+		$admin_email = get_option( 'admin_email' );
+		$sitename    = strtolower( $_SERVER['SERVER_NAME'] );
+		if ( self::pirate_forms_is_localhost() ) {
+			return $admin_email;
+		}
+		if ( substr( $sitename, 0, 4 ) == 'www.' ) {
+			$sitename = substr( $sitename, 4 );
+		}
+		if ( strpbrk( $admin_email, '@' ) == '@' . $sitename ) {
+			return $admin_email;
+		}
+
+		return 'wordpress@' . $sitename;
+	}
+
+	/**
+	 * Get the settings key
+	 *
+	 * @since     1.0.0
+	 */
+	public static function pirate_forms_get_key( $id ) {
+		$pirate_forms_options = get_option( 'pirate_forms_settings_array' );
+		return isset( $pirate_forms_options[ $id ] ) ? $pirate_forms_options[ $id ] : '';
+	}
+
+	/**
+	 * Register the contacts CPT
+	 *
+	 * @since     1.0.0
+	 */
+	public function pirate_forms_register_content_type() {
```

Comment:
@selul this is required for both public and admin, that is why it is defined here